### ISSUE TITLE
Mock.generated doesnt compile

### ISSUE THREAD

**Issue by nskonieczko** [MakeAWishFoundation/SwiftyMocky#260]
after upgrade Property 'someProperty' must be declared public because it matches a requirement in public protocol 'SomeProtocol'

these are not public protocols

**Comment by vadimue** [comment]
Hey @amichnia,
I faced exactly the same issue as @nskonieczko. 
We were using SwiftyMocky 3.5.0, everything was ok, without errors. 
After updating to 4.0.0, we got ~150 errors about `Property 'propertyName' must be declared public because it matches a requirement in public protocol 'ProtocolName'`

When I deleted `Mock.generated.swift` and run `swiftymocky generate` we were still having 7 errors.

Here's a screenshot with errors: https://jmp.sh/v/be8sr5MjJqJWR8S50aaJ

Here's protocol and generated code for it.

Error `Property 'filterGender' must be declared public because it matches a requirement in public protocol 'FiltersStorageProvidable'`
 
```
protocol FiltersStorageProvidable: AutoMockable {
    var filterGender: Gender { get set }
}
```

```
open class FiltersStorageProvidableMock: FiltersStorageProvidable, Mock {
    public init(sequencing sequencingPolicy: SequencingPolicy = .lastWrittenResolvedFirst, stubbing stubbingPolicy: StubbingPolicy = .wrap, file: StaticString = #file, line: UInt = #line) {
        SwiftyMockyTestObserver.setup()
        self.sequencingPolicy = sequencingPolicy
        self.stubbingPolicy = stubbingPolicy
        self.file = file
        self.line = line
    }

    var matcher: Matcher = Matcher.default
    var stubbingPolicy: StubbingPolicy = .wrap
    var sequencingPolicy: SequencingPolicy = .lastWrittenResolvedFirst
    private var invocations: [MethodType] = []
    private var methodReturnValues: [Given] = []
    private var methodPerformValues: [Perform] = []
    private var file: StaticString?
    private var line: UInt?

    public typealias PropertyStub = Given
    public typealias MethodStub = Given
    public typealias SubscriptStub = Given

    /// Convenience method - call setupMock() to extend debug information when failure occurs
    public func setupMock(file: StaticString = #file, line: UInt = #line) {
        self.file = file
        self.line = line
    }

    /// Clear mock internals. You can specify what to reset (invocations aka verify, givens or performs) or leave it empty to clear all mock internals
    public func resetMock(_ scopes: MockScope...) {
        let scopes: [MockScope] = scopes.isEmpty ? [.invocation, .given, .perform] : scopes
        if scopes.contains(.invocation) { invocations = [] }
        if scopes.contains(.given) { methodReturnValues = [] }
        if scopes.contains(.perform) { methodPerformValues = [] }
    }

    var filterGender: Gender {
		get {	invocations.append(.p_filterGender_get); return __p_filterGender ?? givenGetterValue(.p_filterGender_get, "FiltersStorageProvidableMock - stub value for filterGender was not defined") }
		set {	invocations.append(.p_filterGender_set(.value(newValue))); __p_filterGender = newValue }
	}
	private var __p_filterGender: (Gender)?






    fileprivate enum MethodType {
        case p_filterGender_get
		case p_filterGender_set(Parameter<Gender>)

        static func compareParameters(lhs: MethodType, rhs: MethodType, matcher: Matcher) -> Matcher.ComparisonResult {
            switch (lhs, rhs) {            case (.p_filterGender_get,.p_filterGender_get): return Matcher.ComparisonResult.match
			case (.p_filterGender_set(let left),.p_filterGender_set(let right)): return Matcher.ComparisonResult([Matcher.ParameterComparisonResult(Parameter<Gender>.compare(lhs: left, rhs: right, with: matcher), left, right, "newValue")])
            default: return .none
            }
        }

        func intValue() -> Int {
            switch self {
            case .p_filterGender_get: return 0
			case .p_filterGender_set(let newValue): return newValue.intValue
            }
        }
        func assertionName() -> String {
            switch self {
            case .p_filterGender_get: return "[get] .filterGender"
			case .p_filterGender_set: return "[set] .filterGender"
            }
        }
    }

    open class Given: StubbedMethod {
        fileprivate var method: MethodType

        private init(method: MethodType, products: [StubProduct]) {
            self.method = method
            super.init(products)
        }

        static func filterGender(getter defaultValue: Gender...) -> PropertyStub {
            return Given(method: .p_filterGender_get, products: defaultValue.map({ StubProduct.return($0 as Any) }))
        }

    }

    public struct Verify {
        fileprivate var method: MethodType

        public static var filterGender: Verify { return Verify(method: .p_filterGender_get) }
		public static func filterGender(set newValue: Parameter<Gender>) -> Verify { return Verify(method: .p_filterGender_set(newValue)) }
    }

    public struct Perform {
        fileprivate var method: MethodType
        var performs: Any

    }

    public func given(_ method: Given) {
        methodReturnValues.append(method)
    }

    public func perform(_ method: Perform) {
        methodPerformValues.append(method)
        methodPerformValues.sort { $0.method.intValue() < $1.method.intValue() }
    }

    public func verify(_ method: Verify, count: Count = Count.moreOrEqual(to: 1), file: StaticString = #file, line: UInt = #line) {
        let fullMatches = matchingCalls(method, file: file, line: line)
        let success = count.matches(fullMatches)
        let assertionName = method.method.assertionName()
        let feedback: String = {
            guard !success else { return "" }
            return Utils.closestCallsMessage(
                for: self.invocations.map { invocation in
                    matcher.set(file: file, line: line)
                    defer { matcher.clearFileAndLine() }
                    return MethodType.compareParameters(lhs: invocation, rhs: method.method, matcher: matcher)
                },
                name: assertionName
            )
        }()
        MockyAssert(success, "Expected: \(count) invocations of `\(assertionName)`, but was: \(fullMatches).\(feedback)", file: file, line: line)
    }

    private func addInvocation(_ call: MethodType) {
        invocations.append(call)
    }
    private func methodReturnValue(_ method: MethodType) throws -> StubProduct {
        matcher.set(file: self.file, line: self.line)
        defer { matcher.clearFileAndLine() }
        let candidates = sequencingPolicy.sorted(methodReturnValues, by: { $0.method.intValue() > $1.method.intValue() })
        let matched = candidates.first(where: { $0.isValid && MethodType.compareParameters(lhs: $0.method, rhs: method, matcher: matcher).isFullMatch })
        guard let product = matched?.getProduct(policy: self.stubbingPolicy) else { throw MockError.notStubed }
        return product
    }
    private func methodPerformValue(_ method: MethodType) -> Any? {
        matcher.set(file: self.file, line: self.line)
        defer { matcher.clearFileAndLine() }
        let matched = methodPerformValues.reversed().first { MethodType.compareParameters(lhs: $0.method, rhs: method, matcher: matcher).isFullMatch }
        return matched?.performs
    }
    private func matchingCalls(_ method: MethodType, file: StaticString?, line: UInt?) -> [MethodType] {
        matcher.set(file: file ?? self.file, line: line ?? self.line)
        defer { matcher.clearFileAndLine() }
        return invocations.filter { MethodType.compareParameters(lhs: $0, rhs: method, matcher: matcher).isFullMatch }
    }
    private func matchingCalls(_ method: Verify, file: StaticString?, line: UInt?) -> Int {
        return matchingCalls(method.method, file: file, line: line).count
    }
    private func givenGetterValue<T>(_ method: MethodType, _ message: String) -> T {
        do {
            return try methodReturnValue(method).casted()
        } catch {
            onFatalFailure(message)
            Failure(message)
        }
    }
    private func optionalGivenGetterValue<T>(_ method: MethodType, _ message: String) -> T? {
        do {
            return try methodReturnValue(method).casted()
        } catch {
            return nil
        }
    }
    private func onFatalFailure(_ message: String) {
        guard let file = self.file, let line = self.line else { return } // Let if fail if cannot handle gratefully
        SwiftyMockyTestObserver.handleFatalError(message: message, file: file, line: line)
    }
}
```

If I fix all errors manually, project will compile successfully, so it seems those properties really should be public.
For now, we switched to 3.5.0

**Comment by amichnia** [comment]
@vadimue @nskonieczko I finally found the issue. It seems that some time ago we merged code that was fixing a small issue with prototyping. In the case of prototyping, there are no @testable imports, and change added addressed that, making some properties internal.

I'm reverting that change for SwiftyMocky, and I will think of some general solution for this problem. Expect 4.0.1 later today

**Comment by amichnia** [comment]
Resolved with 4.0.1
